File: go/bob/bob.go
package bob

import (
	"regexp"
	"strings"
)

func Hey(input string) string {
	if regexp.MustCompile(`\S+`).MatchString(input) {
		if isYelling(input) {
			return "Woah, chill out!"
		} else if isQuestion(input) {
			return "Sure."
		} else {
			return "Whatever."
		}
	} else {
		return "Fine. Be that way!"
	}
}

func isYelling(input string) bool {
	return strings.ToUpper(input) == input && regexp.MustCompile(`[A-Z]+`).MatchString(input)
}

func isQuestion(input string) bool {
	return regexp.MustCompile(`\?$`).MatchString(input)
}

File: ecmascript/bob/bob.js
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

class Bob {
  hey(message) {
    if (message.match(/^\s*$/)) {
      return 'Fine. Be that way!';
    }

    if (message.match(/[a-zA-Z]/) && message.toUpperCase() === message) {
      return 'Whoa, chill out!';
    }

    if (message.match(/\?$/)) {
      return 'Sure.';
    }

    return 'Whatever.';
  }
}

export default Bob;

